### ISSUE TITLE
Accessing mainWindow in express-app.ts

### ISSUE THREAD

**Issue by hxk1633** [frankhale/electron-with-express#57]
I added a POST endpoint to the express code in this repo and I'd like to be able to send the data from the request body to the renderer process. To achieve this, I need to access the mainWindow object, so I can do something like `windowMain.webContents.send("submit", { data });` inside the POST endpoint. Is this possible? I'm a beginner with electron, so any guidance would be greatly appreciated!

**Comment by frankhale** [comment]
This is a great question. The main process spawns off another instance of electron in node mode that runs the express server, the render process connects to the pages served by the express server. So the main process and render process don't know about the express server because it's in it's own process. There are ways to do it but I don't believe you can gain access the way you described. 

Let me give it some thought.

**Comment by hxk1633** [comment]
@frankhale In this issue https://github.com/frankhale/electron-with-express/issues/20, I think you proposed using Socket.IO for sending messages from express to the renderer. Would this be an alternative solution to achieve what I'm trying to do? If so, is there a minimal example of this somewhere?

**Comment by frankhale** [comment]
Yes, using Socket.IO is very straightforward to do this. I just whipped up an example. See this branch:

https://github.com/frankhale/electron-with-express/tree/socketio-example

**Comment by hxk1633** [comment]
Thanks for taking the time to create an example with socket io. Is there a reason it's in the main process and not a separate process like express? I'm just wondering what performance will be like with this approach.

**Comment by frankhale** [comment]
You mentioned in the original question that you wanted to access the mainWindow. Since it's in the main process you can access the window like you need.

**Comment by frankhale** [comment]
As for performance try it and see how it goes. Performance seems okay to me.

**Comment by frankhale** [comment]
If there is nothing else I'll go ahead and close this. Feel free to reach out if it should still be open.
